core/jazz_azure-create-service/components/ResourceFactory.js

```diff
@@ -0,0 +1,229 @@
+ console.log(JSON.stringify(swaggerString));
+  var parameters = {
+        "contentFormat": "swagger-json",
+        "contentValue": JSON.stringify(swaggerString),
+        "path": basepath
+      };   
+      
+      console.log("parameters here: " + JSON.stringify(parameters));
+ const client = new ApiManagementClient(credentials, subscriptionId);
+ var result = await client.api.createOrUpdateWithHttpOperationResponse(resourceGroupName, serviceName, apiId, parameters, null);
+ return result;
+}
+
+async function deleteApi(resourceGroupName, serviceName, apiId, credentials, subscriptionId) {
+    const client = new ApiManagementClient(credentials, subscriptionId);
+    var result = await client.api.deleteMethodWithHttpOperationResponse(resourceGroupName, serviceName, apiId, "*", null);
+    return result;
+ }
+
+async function addApiToProduct(resourceGroupName, serviceName, productId, apiId, credentials, subscriptionId) {
+    const client = new ApiManagementClient(credentials, subscriptionId);
+    var result = await client.productApi.createOrUpdate(resourceGroupName, serviceName, productId, apiId, null);
+    return result;
+ }
+
+ async function upload(resourceGroup, appName, b64string, subscriptionId, credentials) {
+    var buffer = Buffer.from(b64string, 'base64');
+    var stream = new Stream.PassThrough();
+    stream.end(buffer);
+    
+    const client = await new WebAppManagementClient(credentials, subscriptionId, null, null);
+    var pubcreds = await client.webApps.listPublishingCredentials(resourceGroup, appName, null);
+       console.log(pubcreds);
+        console.log("Trying to upload a file");
+        var config = {
+            headers: {
+                Accept: '*/*'
+              },
+              auth: {
+                username: pubcreds.publishingUserName,
+                password: pubcreds.publishingPassword
+              },
+              encoding: null,
+              body: stream
+        };
+           
+    //   request({
+    //             url: `https://${appName}.scm.azurewebsites.net/api/zipdeploy`,
+    //             method: 'PUT',
+    //             body: stream,
+    //             encoding: null,
+    //             auth: {
+    //                 username: pubcreds.publishingUserName,
+    //                 password: pubcreds.publishingPassword
+    //               },
+    //               headers: {
+    //                 Accept: '*/*'
+    //               }
+    //           }, (error, response) => {
+    //             if (error) {
+    //                 console.log(error);
+    //                 throw error;
+    //             } else {
+    //                 console.log(response);
+    //               return response;
+    //             }
+    //           })
+    
+        return await axios.put(
+            `https://${appName}.scm.azurewebsites.net/api/zipdeploy`,
+            stream,
+            config
```

Comment:
this stream is used twice.  one in this put call, one in the config object --> config.body